### ISSUE TITLE
Feedback

### ISSUE THREAD

**Issue by kcalmes** [deejballa/SCRABBLE#1]
*Some ideas of things you can do to improve your implementation.*

### Create a Word model

This will allow you to demonstrate you can use models and the database. You can use the database migration to seed your csv of words in. You will then be able to use ActiveRecord. You can then make a method on your class that states wether or not a word is valid for the given set of characters. Move as much logic as you can to the model. There is a phrase on the rails community which is "Fat models, skinny controllers" meaning do as little as possible in the controller. Once you construct a good Word model (object) this will happen naturally as you move logic of reading the csv into a migration (or seed script) and move the logic for determining if a word is valid into the model in a method. 

### Set view variables in 1 method

Set all variables that will be used by the view in the index controller. Right now you initialize @result_set on the method you call. Instead return the set, if you still end up with a method like this, and set in the index method.

### Don't use double negative

Your view template is fine, buy I would personally use @result_set.present? instead of the not nil double negative, but that is just a personal style preference I guess.

**Comment by deejballa** [comment]
Thanks a million for your feedback and peer review. 

**Word model**
I was hesitant on creating a model at the beginning because when I had originally been asked to do this I asked him about creating a DB for it and he said to keep it simple this time around so I took that to the extreme and didn't want to make a model at all, but I really felt dirty putting everything in the controller so I decided to take your advice on that one and put the business logic in the Word model. I'm still probably going to wait on creating a database though. 

**View variables**
This makes sense and I had read that, I had everything in the index at first then refactored it into a separate method and at that time forgot to pull it out. I like this best practice though.

**Double negatives**
I didn't even notice the present option. I changed that and it is ready to rock now!

I extracted a couple of simple methods and put them in a StringUtil class just to try to demonstrate the DRY philosophy a little. I also added an exact match option for kicks and giggles. 

LMK if you see anything else or have any other recommendations.
